### ISSUE TITLE
TRY301 flags try - raise - finally block

### ISSUE THREAD

**Issue by jankatins** [charliermarsh/ruff#4287]
[TRY301](https://beta.ruff.rs/docs/rules/raise-within-try/) should flag cases like the below where the same code raises and catches an exception:

```py
try:
     raise Exception()
except:
    .... directly catch the exception
```

On the other hand, the following code is IMO a valid pattern (which would be better handled by a context manager, but in this case, my Consumer does not implement the context manager protocol):

```py
c = confluent_kafka.Consumer(...)
c.subscribe(["topic"])
try:
    # Wait a maximum of 5s for a message: we are doing this in tests
    msg = c.poll(5)
    if msg is None:
        # The next line is flagged with TRY301 Abstract `raise` to an inner function
        raise RuntimeError("Didn't find a message after 5 seconds")
    if msg.error():
        # The next line is flagged with TRY301 Abstract `raise` to an inner function
        raise RuntimeError(f"Error: {msg.error()}")
finally:
    c.close()
```

So from my perspective, the rule should check if there is actually a catching except block...

**Comment by charliermarsh** [comment]
I think that's reasonable.

**Comment by jankatins** [comment]
Thank you!
